task/cosmo-spd/src/main.rs

```diff
@@ -172,13 +180,159 @@ impl idl::InOrderCosmoSpdImpl for ServerImpl {
+    ],
+    [
+        other_sensors::DIMM_C_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_C_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_D_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_D_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_E_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_E_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_F_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_F_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_G_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_G_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_H_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_H_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_I_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_I_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_J_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_J_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_K_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_K_TS1_TEMPERATURE_SENSOR,
+    ],
+    [
+        other_sensors::DIMM_L_TS0_TEMPERATURE_SENSOR,
+        other_sensors::DIMM_L_TS1_TEMPERATURE_SENSOR,
+    ],
+];
+
 impl idol_runtime::NotificationHandler for ServerImpl {
     fn current_notification_mask(&self) -> u32 {
-        0
+        notifications::TIMER_MASK
     }
 
     fn handle_notification(&mut self, _bits: u32) {
-        // Nothing to do here
+        // The FPGA register generation produces different types for bus0 and
+        // bus1, but they're the same shape, so we'll use a small macro for
+        // codegen.
+        macro_rules! dimm_read_temperature {
+            ($addr:ident, $cmd:ident, $count:ident, $data:ident) => {{
+                self.dimms.$cmd.modify(|b| {
+                    b.set_bus_addr($addr);
+                    b.set_len(2);
+                    b.set_reg_addr(0x31); // current sensed temperature
+                    b.set_op(0); // READ
+                });
+                const BUSY_LOOP_COUNT: usize = 32;
+                const TIMEOUT_COUNT: usize = 64;
+                let mut timed_out = false;
+                for i in 0.. {
+                    if self.dimms.$count.data() == 2 {
+                        if i > BUSY_LOOP_COUNT {
+                            ringbuf_entry!(Trace::LoopCount(i));
+                        }
+                        break;
+                    } else if i == TIMEOUT_COUNT {
+                        timed_out = true;
+                        break;
+                    } else if i > BUSY_LOOP_COUNT {
+                        sleep_for(1);
+                    }
+                }
+                if timed_out {
+                    None
+                } else {
+                    Some(self.dimms.$data.data())
+                }
+            }};
+        }
+        let now = sys_get_timer().now;
+        if now >= self.deadline {
```

Comment:
It seems odd to check the deadline here in our notification handler if the timer is our only notification but it's been a while since I thought about hubris timer details

Reply:
I did some reading and believe that we don't need this conditional (or to track the deadline at all), so this has been simplified.